teachertool/src/transforms/removeCriteriaFromChecklist.ts

```diff
@@ -11,10 +11,14 @@ export function removeCriteriaFromChecklist(criteriaInstanceId: string) {
 
     const instance = getCriteriaInstanceWithId(teacherTool, criteriaInstanceId);
     const catalogCriteriaId = instance?.catalogCriteriaId;
+    const allCriteria = [...teacherTool.checklist.criteria];
+    const criteriaIndex = allCriteria.findIndex(c => c.instanceId === criteriaInstanceId);
+    allCriteria[criteriaIndex].deleted = true;
+
 
     const newChecklist = {
         ...teacherTool.checklist,
-        criteria: teacherTool.checklist.criteria.filter(c => c.instanceId !== criteriaInstanceId),
+        criteria: allCriteria,
```

Comment:
So since we're just setting this flag but never removing the criteria, it this means that when a teacher exports their checklist, it will contain all the deleted checklist items which is somewhat wasteful since there's no way to get them back after the toast is dismissed.

I like Eric's suggestion of removing the checklist once the toast is gone. I also think we should move them out of the main criteria and into a separate `deletedCriteria` list, rather than add a delete flag. This means that, even if they export while the toast is open, the export will only contain the un-deleted criteria. It also means we don't need all the deleted checks within the code (which would be easy to forget when implementing future features).

Reply:
Also, if we do a separate list, it's probably still good to remove items once the toast is dismissed, but not quite as important (assuming it clears when the user closes the session).

Reply:
A downside to a separate list for deleted items is that undoing the deletion wouldn't preserve original location in the active criteria list (unless we stored the original index). If undeletion appended to the active set, it may confuse the user, especially if the end of the list is off the page.

Reply:
Hmm, that's true, but I still think it'd be simpler to store a separate list of { criteria + index } objects than have to filter based on deleted everywhere (or remember to call the appropriate helper function)

Reply:
A helper to get active criteria should mitigate this. I agree we don't want to include deleted criteria in the exported JSON.

Reply:
Even with a helper, you have to remember to call it :) Separate list makes it a non-issue as long as we can sort out the index thing. I don't feel that strongly, so I'm happy for Sarah to decide. But personally, I still think a separate list would be slightly cleaner.

Reply:
I would like to continue with the flag approach and use the "get active criteria" helper function. I have been thinking A LOT about the pros and cons of each approach, and I think both have really strong merit. The selling point for me is for the removing and readding scenario of criteria. Both approaches have the same amount of state operations for full removing and newly adding criteria. However, in the new scenario that we are supporting here (removing and readding), the "having a separate list for deleted criteria" would need four state operations along with list operations (two for removing, two for readding) while the flag approach only needs two state operations and no list operations are needed.

Reply:
That's fine, as long as we can get the helper in place and filter deleted criteria out of the serialization.